### ISSUE TITLE
Sort out ReadTheDocs latest vs dev builds

### ISSUE THREAD

**Issue by emiliom** [OSOceanAcoustics/echopype#467]
Moving the topic from https://github.com/OSOceanAcoustics/echopype/issues/378#issuecomment-898778842 to here. Copying what I wrote there (but replacing `master` with `main`):

RTD versions currently set, other than ones corresponding to each release:
- `stable`: built from `stable` branch (per [RTD rules](https://docs.readthedocs.io/en/stable/versions.html#versioned-documentation)). This is the default (https://echopype.readthedocs.io/ redirects to https://echopype.readthedocs.io/en/stable/). See https://readthedocs.org/dashboard/echopype/advanced/
- `latest`: built from `main` ([RTD rules](https://docs.readthedocs.io/en/stable/versions.html#versioned-documentation)). But shouldn't it point to `dev` branch, so it's truly based on the latest development (unreleased) code? Having `latest` based on `main` branch is not terribly useful, relative to the `stable` version.
- `dev` (hidden): built from `dev` branch by default, based on the version name (`dev` version is built from `dev` branch). https://echopype.readthedocs.io/en/dev/ is available but not advertised

I documented this state-of-affairs in the [Contributing to echopype page](https://echopype.readthedocs.io/en/stable/contributing.html#documentation-versions).

And @lsetiawan's comments, plus my response:

> Right now the `dev` branch docs are hidden, so maybe this should be latest, in case someone does want to install the latest and greatest from github, since we have stable pointing to `stable`. 

I agree. Let's point RTD `latest` to the `dev` branch. And then there's no need for the RTD `dev` version any more, so might as well remove it?

It'd be nice to resolve this, but there's also no rush!

**Comment by JessicaS11** [comment]
Excellent docs pages!

Please correct me if I'm misunderstanding, but if you're doing all code/dev and docs patches separately (per [this diagram](https://echopype.readthedocs.io/en/stable/contributing.html#documentation-development), then wouldn't you want `latest` to point to `stable` or `main`? Otherwise, if someone installs from `dev` instead of via a packaged release, the docs they'll automatically see (`latest`, pointed at `dev`) won't match the functionality. Conversely, if they *have* installed a packaged release, they may see docs for functionality that their install doesn't actually have.

**Comment by emiliom** [comment]
> Excellent docs pages!

Thanks! And thanks for your comments, too.

> if you're doing all code/dev and docs patches separately

I wouldn't characterize it that way, exactly. But you're right that that's what the diagram implies; revisiting this now, I also see that we/I never fleshed out a good text description to accompany the diagram. Sigh.

The "code patch" diagram element can and should include corresponding sphinx docs updates, too, if relevant (ie, describing an API change or new functionality). But in practice, such dev docs updates rarely happen at the same time as the code development (sigh), and we typically leave the docs updates to when we're finalizing everything in preparation for a new release. Not of out of a grand design, but simply due to the usual explanations: 24 hours in a day, people are busy and focused.

> Otherwise, if someone installs from `dev` instead of via a packaged release, the docs they'll automatically see (`latest`, pointed at `dev`) won't match the functionality. Conversely, if they *have* installed a packaged release, they may see docs for functionality that their install doesn't actually have.

The docs that are automatically displayed on RTD (the default) are actually `stable`, not `latest`. We rigged it that way so that the most readily visible docs (the default) align with the currently released package version while incorporating incremental improvements in the docs coming in from the GitHub `stable` branch. A user would have to specifically select the `latest` RTD docs version to see docs that correspond to the 'main` branch, or otherwise know about the existence of the hidden `dev` version from the Contributing to echopype docs page to see docs that match the 'dev' branch. I guess the assumption is that someone who is installing from `dev` knows that they're in fragile territory and can be expected to act accordingly (eg, dig around, read the Contributing to echopype page, ask questions, expect problems, etc).

The focus of this issue is specifically on whether to change our current setup vis a vis the RTD `latest` vs `dev` versions. I think it's unnecessarily complex, the `latest` version is not serving a useful purpose, and `dev` is too hidden. The proposal is to drop RTD `dev` and point `latest` to the GitHub `dev` branch instead.

**Comment by JessicaS11** [comment]
> revisiting this now, I also see that we/I never fleshed out a good text description to accompany the diagram. Sigh.

If you ping me, I'd be happy to look quickly at the updated diagram.

>  But in practice, such dev docs updates rarely happen at the same time as the code development (sigh), and we typically leave the docs updates to when we're finalizing everything in preparation for a new release. Not of out of a grand design, but simply due to the usual explanations: 24 hours in a day, people are busy and focused.

I can relate to that challenge (for me the docs seem to make it more often with the code, but the testing is what struggles to keep up)! I'll let you know when I figure out how to be in multiple places at once (but don't hold your breath!). : )

> The docs that are automatically displayed on RTD (the default) are actually `stable`, not `latest`. We rigged it that way so that the most readily visible docs (the default) align with the currently released package version while incorporating incremental improvements in the docs coming in from the GitHub `stable` branch.

Thanks for the added explanation. I think your approach to point `latest` to `dev` and drop [duplicate, hidden] `dev` branch builds on RTD makes a lot of sense. Perhaps you could include some sort of label indicating the docs build pointers on the diagram? I can't remember how transparent this info is to non-admins on RTD.

**Comment by emiliom** [comment]
> If you ping me, I'd be happy to look quickly at the updated diagram.

Thanks! I've prepared an update where the diagram stays the same but I've added text explanations. I haven't pushed it out yet.

> Thanks for the added explanation. I think your approach to point latest to dev and drop [duplicate, hidden] dev branch builds on RTD makes a lot of sense. Perhaps you could include some sort of label indicating the docs build pointers on the diagram? I can't remember how transparent this info is to non-admins on RTD.

Thanks for the feedback. I agree that making this information more explicit would be helpful. I tried just now modifying the diagram to indicate the matching RTD versions, but decided that it added too much clutter and complicated the diagram.

**Comment by emiliom** [comment]
We've moved forward with this! I've changed the RTD `latest` version to point to the `dev` branch. Here's an RTD admin screenshot of where this is done (see "Default branch"). The next (and last) step is to [edit the corresponding echopype documentation section](https://echopype.readthedocs.io/en/stable/contributing.html#documentation-versions) that described what RTD versions we have. 

![image](https://user-images.githubusercontent.com/742403/149416409-a14afc4b-be92-482b-8496-d0aec4656eaa.png)
